utils/build-osx.sh

```diff
@@ -43,6 +44,97 @@ function create_virtualenv_for_pyinstaller {
     pip3 install -e ".[test]"
 }
 
+# Overriding this function to deal with the x86 special case
+function make_hash_if_necessary {
+    cd pyinstaller/electron
+    echo "    --> calculate the hash of the binary for download"
+    specterd_plt_filename=../dist/${specterd_filename}
+    # early exit
+    if [[ "$make_hash" != 'True' ]]; then
+      node ./set-version $version
+      return 0
+    fi
+    # Making the hash only makes sense on a arm arch
+    if [[ "$ARCH" != "arm64" ]]; then
+      echo "ERROR: make-hash should be only called on an arm64 machine on a mac"
+      exit 1
+    fi
```

Comment:
move check downwards and also check for electron target, better comment!

Reply:
fixed in [b8a3a7c](https://github.com/cryptoadvance/specter-desktop/pull/2446/commits/b8a3a7ce4ef1e1dfedd24e856d765754f1d5843f)